Feat: Update e2e test cases according to the update of backend

Jerify02 opened the pull request:
Close #6523, #6628 .

marlass commented on projects/storefrontapp-e2e-cypress/cypress/helpers/notification.ts:
```diff
@@ -218,13 +207,30 @@ export function navigateToPDPInCustomerInterest(productCode: string) {
   cy.get('.cx-product-interests-product-item').should('have.length', 10);
-  cy.get('cx-pagination:first .page-link').should('have.length', 4);
+  cy.get('cx-pagination:first a').should('have.length', 4);
+  cy.get('cx-pagination:first a')
```
This fails in mobile e2e test, beause top pagination is hidden on mobile

Jerify02 replied:
Fixed. Thanks!

Jerify02 commented:
> Recent modifications with handling of quantities now allow for different handling of entry removal.
> You could use the same mechanism we use to update entry quantity. You have information if that is the selective cart in `getControl` method. You could place the whole logic there and drop any changes you made with new input and output. It would be much simpler and easier to maintain.

When I added this modifications, the `Move to Cart` & `Remove` button in saveForLater are disabled, so I added this changes, and now it works fine(maybe the bug was fixed).
